flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/RecordWriterTest.java

```diff
@@ -210,10 +354,12 @@ public void testBroadcastEventNoRecords() throws Exception {
 		}
 
 		TestPooledBufferProvider bufferProvider = new TestPooledBufferProvider(Integer.MAX_VALUE, bufferSize);
-
 		ResultPartitionWriter partitionWriter = new CollectingPartitionWriter(queues, bufferProvider);
-		RecordWriter<ByteArrayIO> writer = new RecordWriter<>(partitionWriter);
-		CheckpointBarrier barrier = new CheckpointBarrier(Integer.MAX_VALUE + 919192L, Integer.MAX_VALUE + 18828228L, CheckpointOptions.forCheckpointWithDefaultLocation());
+		RecordWriter<?> writer = isBroadcastWriter ?
+			RecordWriter.createRecordWriter(partitionWriter, new OutputEmitter(ShipStrategyType.BROADCAST, 0), "test") :
+			RecordWriter.createRecordWriter(partitionWriter);
+
+		CheckpointBarrier barrier = new CheckpointBarrier(0, 0, CheckpointOptions.forCheckpointWithDefaultLocation());
```

Comment:
Just for simplify the codes. :) 
This line is so long because of the `CheckpointID` and timeout info which seem not helpful for the test function, so I refer to other tests to change them to 0 simple.